desktop: work around client-side rounding issues at right/bottom pixel

jlindgren90 opened the pull request:
Tested with qmpanel (similar to lxqt-panel) at 1x and 2x output scale. Does not help if scaling is done client-side, e.g. QT_SCALE_FACTOR=2.

Fixes: #2379
Fixes: #3099

tokyo4j commented on src/desktop.c:
```diff
@@ -269,6 +269,35 @@ get_cursor_context(struct server *server)
+		 * versions (likely to be fixed soon in wlroots). See:
+		 *
+		 * - https://gitlab.freedesktop.org/wayland/wayland/-/issues/555
+		 * - https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/5159
```
nit: I think this paragraph should be moved to the commit message, because "soon" will no longer be "soon" as time passes.

jlindgren90 replied:
Sure, I was on the fence whether to mention it here in the comment or not.

jlindgren90 commented:
We have a number of workarounds for client-side issues in place already. Some even for specific clients (check xdg.c for references to Thunderbird).

Obviously there have to be some boundaries, but I think in the past we've leaned toward pragmatism and added workarounds where it fixes a popular use-case and doesn't have bad side effects.

This particular one is important for LXQt, and the issue is not easy to fix in Qt (would require replacing a common but incorrect usage of QRect::contains throughout the codebase).

jlindgren90 commented:
- moved to a separate function (edge_rounding_workaround)
- expanded rationale in comments a bit
- moved mention of the wlroots/wayland issue to the commit message

jlindgren90 commented:
- renamed new function to `avoid_edge_rounding_issues`

johanmalm commented:
This is great. Thank you so much for investigating + fixing.
